File: algorithms/0088-merge-sorted-array.go
package algorithms

func merge(nums1 []int, m int, nums2 []int, n int)  {
	nums3 := []int{}
	a, b := 0, 0
	for a < m && b < n {
		if nums1[a] < nums2[b] {
			nums3 = append(nums3, nums1[a])
			a++
		} else {
			nums3 = append(nums3, nums2[b])
			b++
		}
	}
	for a < m {
		nums3 = append(nums3, nums1[a])
		a++
	}
	for b < n {
		nums3 = append(nums3, nums2[b])
		b++
	}
	for i := 0; i < m + n; i++ {
		nums1[i] = nums3[i]
	}
}

File: algorithms/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        a, b = m - 1, n - 1
        while a >= 0 and b >= 0:
            if nums1[a] > nums2[b]:
                nums1[a + b + 1] = nums1[a]
                a -= 1
            else:
                nums1[a + b + 1] = nums2[b]
                b -= 1
        while b >= 0:
            nums1[b] = nums2[b]
            b -= 1
